[Merged by Bors] - chore: remove useless tactics

adomani opened the pull request:
The removal of some pointless tactics flagged by #11308.

---
<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

To indicate co-authors, include lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]
-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

grunweg commented on Mathlib/Data/Ordmap/Ordset.lean:
```diff
@@ -395,7 +395,8 @@ theorem node3R_size {l x m y r} : size (@node3R α l x m y r) = size l + size m
 theorem node4L_size {l x m y r} (hm : Sized m) :
     size (@node4L α l x m y r) = size l + size m + size r + 2 := by
-  cases m <;> simp [node4L, node3L, node'] <;> [skip; simp [size, hm.1]] <;> abel
+
```
Extra newline?

adomani replied:
Yes, thanks!  Removed the extra line!

grunweg reviewed (approved):
I love the smell of deleted code in the evening. I'm amazed that new linters can produce :-)

adomani commented:
> I love the smell of deleted code in the evening. I'm amazed what gains new linters can yield :-)

Honestly, I was not expecting that so many `skip`s and various other unused tactics would be present...

YaelDillies commented on Mathlib/CategoryTheory/Limits/FunctorCategory.lean:
```diff
@@ -321,7 +321,7 @@ theorem colimit_obj_ext {H : J ⥤ K ⥤ C} [HasColimitsOfShape J C] {k : K} {W
 instance evaluationPreservesLimits [HasLimits C] (k : K) :
     PreservesLimits ((evaluation K C).obj k) where
-  preservesLimitsOfShape {J} 𝒥 := by skip; infer_instance
+  preservesLimitsOfShape {_} _𝒥 := inferInstance
```
Doesn't `inferInstance` add extra junk that `by infer_instance` doesn't? Can you check the term is syntactically the same as it was before?

adomani replied:
In its current form (with `inferInstance`):
```lean
#print evaluationPreservesLimits

def CategoryTheory.Limits.evaluationPreservesLimits.{v₂, u₂, v, u} : {C : Type u} →
  [inst : Category.{v, u} C] →
    {K : Type u₂} →
      [inst_1 : Category.{v₂, u₂} K] → [inst_2 : HasLimits C] → (k : K) → PreservesLimits ((evaluation K C).obj k) :=
fun {C} [Category.{v, u} C] {K} [Category.{v₂, u₂} K] [HasLimits C] k ↦
  { preservesLimitsOfShape := fun {x} _𝒥 ↦ inferInstance }
```
As it was before the change:
```lean
#print evaluationPreservesLimits

def CategoryTheory.Limits.evaluationPreservesLimits.{v₂, u₂, v, u} : {C : Type u} →
  [inst : Category.{v, u} C] →
    {K : Type u₂} →
      [inst_1 : Category.{v₂, u₂} K] → [inst_2 : HasLimits C] → (k : K) → PreservesLimits ((evaluation K C).obj k) :=
fun {C} [Category.{v, u} C] {K} [Category.{v₂, u₂} K] [HasLimits C] k ↦
  { preservesLimitsOfShape := fun {J} 𝒥 ↦ inferInstance }
```
Seems that the only difference are the variable names.

YaelDillies commented on Mathlib/Data/Ordmap/Ordset.lean:
```diff
@@ -395,7 +395,7 @@ theorem node3R_size {l x m y r} : size (@node3R α l x m y r) = size l + size m
 theorem node4L_size {l x m y r} (hm : Sized m) :
     size (@node4L α l x m y r) = size l + size m + size r + 2 := by
-  cases m <;> simp [node4L, node3L, node'] <;> [skip; simp [size, hm.1]] <;> abel
+  cases m <;> simp [node4L, node3L, node'] <;> [abel; (simp [size, hm.1]; abel)]
```
Not sure that one is necessarily an improvement

adomani replied:
I am happy to revert it back to where it was, but it is shorter and produces the same proof.  (Really identical, very long, output of `#print`.)

YaelDillies commented on Mathlib/LinearAlgebra/LinearIndependent.lean:
```diff
@@ -518,8 +518,7 @@ theorem linearIndependent_iUnion_of_directed {η : Type*} {s : η → Set M} (hs
     (h : ∀ i, LinearIndependent R (fun x => x : s i → M)) :
     LinearIndependent R (fun x => x : (⋃ i, s i) → M) := by
   by_cases hη : Nonempty η
```
You will be able to golf further by doing `cases isEmpty_or_nonempty η` instead of that `by_cases

adomani replied:
Ok, but I would definitely say that this is out of scope for this PR!

YaelDillies commented on Mathlib/SetTheory/Game/Short.lean:
```diff
@@ -205,9 +205,7 @@ instance listShortGet :
-    rcases n with ⟨_, ⟨⟩⟩
-    -- Porting note: The proof errors unless `done` or a `;` is added after `rcases`
-    done
+    rcases n with ⟨_, ⟨⟩⟩;
```
Seems like you haven't addressed the porting note, and removing `done` isn't a particularly great idea either.

I suggest you nolint `done` calls by default.

adomani replied:
Isn't the issue that `rcases` does not realize that its pattern is over, since `|` could be an `or`?  Again, I can put the porting note back, but replacing `done` by `;` also "fixes" the error.

adomani replied:
Here is an example:
```lean
example (x : Bool) (h : false ∨ true) : True := by
  match x with
    | true => rcases h with ⟨⟩
    | false  -- <--- this is still part of `rcases`: I am calling `false` the `true` proof! x)
              exact .intro  -- <--- the proof if `h` is `false`
              exact .intro  -- <--- the proof if `h` is `true`
    | false => exact .intro -- <--- the second `match` arm!

```

YaelDillies reviewed (commented):
!bench

riccardobrasca commented:
!bench

YaelDillies commented:
Seems like some tactics weren't useless after all.

adomani commented:
:facepalm: the file that slowed down is a file where I replaced some splits and assumptions into a single `aesop` call.  I reverted that change, since it was only flagged up by a `skip` and it should not be flagged by the second generation linter anyway!

adomani commented:
!bench

riccardobrasca commented:
Thanks!

bors merge
